### DISCUSSION TITLE
Firebase Messaging versioning issue when converting from Cordova to Capacitor

### DISCUSSION THREAD

**Discussion by roryjarrard** [ionic-team/capacitor#5657] [General] [answered]
We previously were using the Firebasex plugin in our Cordova application. We converted to Capacitor, removed the Firebasex plugin, and installed `@capacitor/push-notifications`.  However, the documentation to implement push notifications (https://capacitorjs.com/docs/guides/push-notifications-firebase#prepare-an-ionic-capacitor-app) can't be followed because the CordovaPluginsStatic.podspec is requiring the dependency of FirebaseMessaging 2.0.0, when the latest is 9.1.0. 

When running `npx cap sync`:

```
✖ update ios - failed!
[error] Analyzing dependencies
        [!] CocoaPods could not find compatible versions for pod "FirebaseMessaging":
        In Podfile:
        CordovaPluginsStatic (from `../capacitor-cordova-ios-plugins`) was resolved to 3.5.1, which depends on
        FirebaseMessaging (~> 2.0.0)
        
        Firebase/Messaging was resolved to 9.1.0, which depends on
        FirebaseMessaging (~> 9.1.0)

```

I can't get past this, can't update the pods because of this, but have no idea where this dependency is being generated. I've gone to extremes, even trying to remove that spec but of course it just shows up again. 

I thought maybe another Cordova plugin was causing this but there are no references to Firebase in the plugins folder. How can I get this to update to a later version of FirebaseMessaging?

**Comment by jcesarmobile** [comment, marked as answer]
That dependency comes from a cordova plugin, looks like you didn’t uninstall the firebasex plugin correctly or have another plugin using firebase dependencies.

**Reply by roryjarrard** [reply]
Not a Cordova specific plugin, but `phonegap-plugin-push@^2.3.0` was installed which was the culprit. Thanks.
